### PR TITLE
Update main.py and allow the ability to choose the amount of goals the user wants instead of 5.

### PR DESCRIPTION
**Pull request by lightlord1233**
Choosing the amount of goals. Say the user wants 3 goals, or 10 goals.

### CODE REVIEW

File: scripts/main.py [hunk 1]
```diff
@@ -219,14 +219,29 @@ def prompt_user():
     if ai_role == "":
         ai_role = "an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth."
 
-    # Enter up to 5 goals for the AI
+    # Enter goals for the AI
     print_to_console(
-        "Enter up to 5 goals for your AI: ",
+        "How many goals do you want the AI to have? (Please enter a number between 1 and 10): ",
```
Comment (LumiroCode) [inline scripts/main.py]:
```suggestion
        "Enter how many goals do you want the AI to have: ",
```

File: scripts/main.py [hunk 2]
```diff
@@ -219,14 +219,29 @@ def prompt_user():
...
         ai_role = "an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth."
 
-    # Enter up to 5 goals for the AI
+    # Enter goals for the AI
     print_to_console(
-        "Enter up to 5 goals for your AI: ",
+        "How many goals do you want the AI to have? (Please enter a number between 1 and 10): ",     
         Fore.GREEN,
-        "For example: \nIncrease net worth, Grow Twitter Account, Develop and manage multiple businesses autonomously'")
+        "For example, 'An Number between 1 to 10")
```
Comment (LumiroCode) [inline scripts/main.py]:
```suggestion
        "Please enter a number between 1 and 10")
```

**Comment by lightlord1233** [comment]
Perhaps, but after a certain amount of goals it is likely too long to follow the goals, hence the range.

> Currently, you can enter <=5 goals, the more flexible way would probably be to just allow it to be entirely dynamic. As in, just keep prompting the user for goals until they've entered an empty goal.

Perhaps, but after a certain amount of goals it is likely too long to follow the goals, hence the range.

File: scripts/main.py [hunk 3]
```diff
@@ -219,14 +219,29 @@ def prompt_user():
...
+ 
+    while True:
+        num_goals = input("")
+        if num_goals.isdigit() and 1 <= int(num_goals) <= 10:
+            num_goals = int(num_goals)
+            break
+        else:
+            print_to_console(
+                "Please enter a valid number between 1 and 10: ",
```
Comment (LumiroCode) [inline scripts/main.py]:
```suggestion
                "Please enter a valid number between 1 and 10: ",
```

File: scripts/main.py [hunk 4]
```diff
@@ -219,14 +219,29 @@ def prompt_user():
...
+ 
+    while True:
+        num_goals = input("")
+        if num_goals.isdigit() and 1 <= int(num_goals) <= 10:
+            num_goals = int(num_goals)
+            break
+        else:
+            print_to_console(
+                "Please enter a valid number between 1 and 10: ",     
+                Fore.RED)
```
Comment (LumiroCode) [inline scripts/main.py]:
```suggestion
                Fore.RED,
                "")
```

**Review by ryanmac — DISMISSED** [review]
This looks good to me.

**Review by lightlord1233 — COMMENTED** [review]
Approved most of the changes except Line 236 change as its extra code with no benefit.

**Comment by Gerqus** [comment]
> Approved most of the changes except Line 236 change as its extra code with no benefit.

It's suggestion to comply with method defined interface - it expects one more argument. Don't you think that not passing a required argument make code more bug-prone?. I know it's minor here x) I just have this bias towards fulfilling interfaces as defined for more robust code. But I understand your point of view as well :+1:

**Pull request closed without merging**